src/app/signers/SignersTable.tsx

```diff
@@ -305,48 +283,63 @@ const SignersTableBase = () => {
     [setSelectedCycle]
   );
 
-  const {
-    data: { results: signers },
-  } = useSuspensePoxSigners(selectedCycle);
+  const signersResponse = useSuspensePoxSigners(parseInt(selectedCycle), {
+    limit: 100,
+  });
+  const signers = useSuspenseInfiniteQueryResult<PoxSigner>(signersResponse);
 
   if (!signers) {
     throw new Error('Signers data is not available');
   }
 
-  const queryClient = useQueryClient();
-  const getSignerMetricsBySignerQuery = useGetSignerMetricsBySignerQuery();
-  const signersMetricsQueries = useMemo(() => {
-    return {
-      queries: signers.map(signer => {
-        return getSignerMetricsBySignerQuery(parseInt(selectedCycle), signer.signing_key);
-      }),
-      combine: (response: UseQueryResult<SignerMetricsSignerForCycle, Error>[]) =>
-        response.map(r => r.data ?? ({} as SignerMetricsSignerForCycle)),
-    };
-  }, [signers, getSignerMetricsBySignerQuery, selectedCycle]);
-  const signersMetrics = useQueries(signersMetricsQueries, queryClient);
+  const signersMetricsResponse = useSignerMetricsSignersForCycle(parseInt(selectedCycle), {
+    limit: 100,
+  });
 
-  const signersData = useMemo(
-    () =>
-      signers
-        .map((signer, index) => {
-          return {
-            ...formatSignerRowData(signer, signersMetrics[index]),
-          };
-        })
-        .sort((a, b) =>
-          votingPowerSortOrder === 'desc'
-            ? b.votingPower - a.votingPower
-            : a.votingPower - b.votingPower
-        ),
-    [signers, signersMetrics, votingPowerSortOrder]
-  );
+  const signersMetrics =
+    useInfiniteQueryResult<SignerMetricsSignerForCycle>(signersMetricsResponse);
+
+  const signersData = useMemo(() => {
+    return signers
+      .map(signer => {
+        const metrics =
+          signersMetrics.find(m => m.signer_key === signer.signing_key) ||
+          ({} as SignerMetricsSignerForCycle);
```

Comment:
If metrics is passed as empty object to formatSignerRowData it'll cause totalProposals to be NaN

Reply:
this is handled here 
export function formatSignerProposalMetric(metric: number): string {
  if (isNaN(metric)) return '-';
  if (metric === 0) return '0%';
  if (metric === 1) return '100%';
  return `${(metric * 100).toFixed(1)}%`;
}